_extras/miscellaneous.md

```diff
@@ -170,6 +170,135 @@ outputs:
     type: File
     outputSource: first/txt
 ```
+### Optional Inputs 💯
+
+In CWL unlike WDL optional input settings are defined in the type section of the yaml and not on the declaration. Notice the question mark is on the type.
+
+```yaml
+inputs:
+  InputRead1:
+    type: File
+    inputBinding:
+      position: 100
+
+  #Optional Inputs
+  isCasava:
+    type: boolean?
+    inputBinding:
+      position: 1
+      prefix: "--casava"
+```
+<a name="enuminputs"></a>
+### Enum Inputs ⚜️
```

Comment:
This is defined at https://www.commonwl.org/user_guide/19-custom-types/index.html

Maybe it deserves its own mini-lesson?

Reply:
I would say a mini-lesson would be good, this was a boiled down version for all of us to reference quickly.